components/layout/fragment.rs

```diff
@@ -921,7 +922,34 @@ impl Fragment {
             specific: specific,
             inline_context: None,
             pseudo: node.get_pseudo_element_type().strip(),
-            flags: FragmentFlags::empty(),
+            flags: FragmentFlags::with_quirks_mode(node.quirks_mode()),
+            debug_id: DebugId::new(),
+            stacking_context_id: StackingContextId::new(0),
+        }
+    }
+
+    /// Constructs a new `Fragment` from a node, overriding its style
+    pub fn from_node_and_style<N: ThreadSafeLayoutNode>(node: &N,
+                                                        style: Arc<ServoComputedValues>,
+                                                        specific: SpecificFragmentInfo)
+                                                        -> Fragment {
```

Comment:
I created this to simplify some of the existing code, but it's not _really_ necessary for this patch. Tell me if you'd prefer that I revert the changes around this constructor.

If you decide that we should keep it, maybe `new` could call it too and cut some code.